### PR TITLE
Refactor `cudf::detail::contains(table_view, table_view)`

### PR DESCRIPTION
**Pull request by ttnghia**
This PR modifies the overload of `cudf::detail::contains` that accepts a pair of `table_view`. The main change here is switching to using `cuco::static_map` instead of `cuco::static_multimap`. This can avoid the performance regression due to the limitation of using multimap as has been reported in https://github.com/rapidsai/cudf/issues/11299, when the input tables have many duplicate rows.

### CODE REVIEW

File: cpp/src/search/contains_table.cu [hunk 1]
```diff
@@ -36,6 +36,60 @@ namespace {
...
+ * @brief An adapter functor to support strong index types for row hasher.
+ */
+template <typename Hasher>
+struct strong_index_hasher_adapter {
+  strong_index_hasher_adapter(Hasher const& hasher) : _hasher{hasher} {}
+
+  template <typename T>
+  __device__ inline auto operator()(T const idx) const noexcept
+  {
+    return _hasher(static_cast<size_type>(idx));
```
Comment (PointKernel) [inline cpp/src/search/contains_table.cu:49]: Will there be narrow conversion issues?

**Review by PointKernel — APPROVED** [review]
LGTM

File: cpp/src/search/contains_table.cu (hunk 1 above)
Comment (ttnghia) [reply cpp/src/search/contains_table.cu:49]: This functor is only used locally in this file, and the input `idx` is only strong index types (either `lhs_index_type` or `rhs_index_type`).

**Comment by mythrocks** [comment]
This was looking good to me. I'd like to get my head around the null-checking before I approve.

**Comment by hyperbolic2346** [comment]
Do we have any performance results for this yet?

**Comment by ttnghia** [comment]
Close this as it is covered in a new PR: https://github.com/rapidsai/cudf/pull/11330.

**Pull request closed without merging**